src/login/oidc/TokenRequester.ts

```diff
@@ -40,24 +42,32 @@ export default class TokenRequester {
     private issuerConfigFetcher: IIssuerConfigFetcher,
     @inject("fetcher") private fetcher: IFetcher,
     @inject("dpopHeaderCreator") private dpopHeaderCreator: IDpopHeaderCreator,
-    @inject("joseUtility") private joseUtility: IJoseUtility
+    @inject("joseUtility") private joseUtility: IJoseUtility,
+    @inject("clientRegistrar") private clientRegistrar: IClientRegistrar,
+    @inject("dpopClientKeyManager")
+    private dpopClientKeyManager: IDpopClientKeyManager
   ) {}
 
   async request(
-    localUserId: string,
+    sessionId: string,
     body: Record<string, string>
   ): Promise<void> {
-    const [issuer, clientId, clientSecret] = await Promise.all([
-      this.storageUtility.getForUser(localUserId, "issuer", true),
-      this.storageUtility.getForUser(localUserId, "clientId", true),
-      this.storageUtility.getForUser(localUserId, "clientSecret")
+    const [issuer] = await Promise.all([
+      this.storageUtility.getForUser(sessionId, "issuer", {
+        errorIfNull: true
```

Comment:
Strange name - throw error if what is Null? Seems like it could have a more descriptive, helpful name...

Reply:
That's a generic parameter in the storage utility, which controls if an error should be thrown if the requested key does not exist. I'll create a ticket to refactor this.